pkg/ocm/share/repository/nextcloud/nextcloud.go

```diff
@@ -153,27 +162,47 @@ func (sm *Manager) StoreShare(ctx context.Context, share *ocm.Share) (*ocm.Share
 // GetShare gets the information for a share by the given ref.
 func (sm *Manager) GetShare(ctx context.Context, user *userpb.User, ref *ocm.ShareReference) (*ocm.Share, error) {
 	data, err := json.Marshal(ref)
+	log := appctx.GetLogger(ctx)
 	if err != nil {
 		return nil, err
 	}
-	_, body, err := sm.do(ctx, Action{"GetShare", string(data)}, getUsername(user))
+	_, body, err := sm.do(ctx, Action{"GetSentShareByToken", string(data)}, getUsername(user))
```

Comment:
Here the call expects to return a share either by ID or by Token, that is, `data` here may contain any of those values (see `ocm.ShareReference`). @MahdiBaghbani how is this implemented on the PHP side in the RevaController? Is this endpoint handling all cases or just the GetShareByToken as the name suggests?

For instance, https://github.com/cs3org/reva/blob/updated-sciencemesh-testing/internal/grpc/services/gateway/ocmshareprovider.go#L174-L180 is a request by ID.

Reply:
GetSentShareByToken doesn't handle ID (or is it really `opaque_id` ?), so we have to change the function if it should also expect an ID.

Might be relevant to this:
https://github.com/pondersource/sciencemesh-php/issues/208

Reply:
The panic in https://github.com/pondersource/sciencemesh-php/issues/208 was solved, but the issue remains, you do need to implement a GetShareById (and you could just call the PHP endpoint `GetShare`, with different payloads). The ID there is whichever ID was stored in the database, it's the `providerID` as it is named in the [OCM payload](https://cs3org.github.io/OCM-API/docs.html?branch=develop&repo=OCM-API&user=cs3org#/paths/~1shares/post)

Reply:
Also another difference is that it now calls `getUsername(user)` instead of `nobody`, Which needs to be reflected on PHP side too

Reply:
ah right, hmm I think then that it's ok to keep `nobody` explicitly: here the user is remote, and `getUsername()` will actually return `nobody` or `''`

Reply:
Eventually I fixed the return value of `getUsername()` when the user can't be resolved, as it is used in all other calls (and hardcoding `nobody` really felt like a temporary hack)